code/datums/extensions/lockable.dm

```diff
@@ -0,0 +1,218 @@
+/datum/extension/lockable
+	base_type = /datum/extension/lockable
+	expected_type = /obj/item
+	flags = EXTENSION_FLAG_IMMEDIATE
+	// Configuration variables
+	var/max_code_length = 5		// Max length of the password.
+
+	// Stateful variables
+	var/locked 			= 1		// Is it locked?
+	var/code 					// The passcode currently inputed.
+	var/l_code 			= null	// The actual code of the lock
```

Comment:
You are assigning null here and not bothering with the var above - should probably go with not bothering for both for consistency's sake.